File: spring-flyway/src/main/java/com/example/flyway/domain/Member.java
package com.example.flyway.domain;

import lombok.Getter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

import static javax.persistence.GenerationType.IDENTITY;

@Entity
@Getter
public class Member {

    @Id @GeneratedValue(strategy = IDENTITY) @Column(name = "member_id")
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(length = 100)
    private String intro;
}

File: spring-events/src/main/kotlin/com/spring/events/domain/Member.kt
package com.spring.events.domain

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import org.springframework.data.domain.AbstractAggregateRoot

@Entity
class Member(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    var name: String,
    var age: Int,
) : AbstractAggregateRoot<Member>() {
    val requiredId: Long
        get() = requireNotNull(id) { "not persisted !" }

    val firstIssueCouponName: String
        get() = "$name 님의 회원 가입 쿠폰"

    private fun publishEvent(event: RegisterMemberEvent) {
        registerEvent(event)
    }

    companion object {
        fun register(command: RegisterMember) = Member(
            name = command.name,
            age = command.age,
        )
    }
}

data class RegisterMemberEvent(
    val memberId: Long,
)

data class RegisterMember(
    val name: String,
    val age: Int,
)
